src/lib/stores/data-converter-websocket.ts

```diff
@@ -0,0 +1,43 @@
+import { get } from 'svelte/store';
+import WebSocketAsPromised from 'websocket-as-promised';
+import {
+  dataConverterPort,
+  setLastDataConverterFailure,
+} from './data-converter-config';
+
+export const dataConverterWebsocket = (function () {
+  let sock = null;
+  const port = get(dataConverterPort) ?? null;
+
+  if (port == null) {
+    return {
+      websocket: function () {
+        return null;
+      },
+      closeSocket: function () {
+        return null;
+      },
+    };
+  }
+
+  try {
+    sock = new WebSocketAsPromised(`ws://localhost:${port}/`, {
+      packMessage: (data) => JSON.stringify(data),
+      unpackMessage: (data) => JSON.parse(data as string),
+      attachRequestId: (data, requestId) =>
+        Object.assign({ requestId: requestId }, data),
+      extractRequestId: (data) => data && data.requestId,
+    });
+  } catch (err) {
+    setLastDataConverterFailure();
+  }
+
+  sock.open();
+
+  return {
```

Comment:
So, to my previous comment, You can just export this object and it’s the same as the IIFE.